distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso

```diff
@@ -348,82 +346,51 @@ Text.find pattern mode=Mode.All match_ascii=Nothing case_insensitive=Nothing dot
+   - matcher: If a `Text_Matcher`, the text is compared using case-sensitivity
+     rules specified in the matcher. If a `Regex_Matcher`, the term is used as a
+     regular expression and matched using the associated options.
 
    > Example
-     Split the comma-separated text into a vector of items.
+     Split the text on any occurrence of the separator `"::"`.
 
-         "ham,eggs,cheese,tomatoes".split ","
+         example_split =
+             text = "Namespace::package::package::Type"
+             text.split "::" == ["Namespace", "package", "package", "Type"]
 
    > Example
-     Split the text on whitespace into a vector of items.
+     Split the text on a regex pattern.
 
-         "ham eggs cheese tomatoes".split Split_Kind.Whitespace
+         "abc--def==>ghi".split "[-=>]+" Regex_Matcher == ["abc", "def", "ghi"]
 
    > Example
-     Split the text on any occurrence of the separator `"::"`.
+     Split the text on any whitespace.
 
-         example_split =
-             text = "Namespace::package::package::Type"
-             text.split ":::"
-Text.split : Split_Kind -> Mode.Mode -> Boolean | Nothing -> Boolean | Nothing -> Boolean | Nothing -> Boolean | Nothing -> Boolean | Nothing -> Vector.Vector Option.Option -> Vector.Vector Text
-Text.split separator=Split_Kind.Whitespace mode=Mode.All match_ascii=Nothing case_insensitive=Nothing dot_matches_newline=Nothing multiline=Nothing comments=Nothing extra_opts=[] =
-    case separator of
-        Split_Kind.Words -> Vector.Vector this.words
-        Split_Kind.Whitespace ->
-            pattern = Regex.compile "\s+" match_ascii=match_ascii case_insensitive=case_insensitive dot_matches_newline=dot_matches_newline multiline=multiline comments=comments extra_opts=extra_opts
-            pattern.split this mode=mode
-        Split_Kind.Lines ->
-            pattern = Regex.compile "\v+" match_ascii=match_ascii case_insensitive=case_insensitive dot_matches_newline=dot_matches_newline multiline=multiline comments=comments extra_opts=extra_opts
-            pattern.split this mode=mode
-        Text ->
-            pattern = Regex.compile separator match_ascii=match_ascii case_insensitive=case_insensitive dot_matches_newline=dot_matches_newline multiline=multiline comments=comments extra_opts=extra_opts
-            pattern.split this mode=mode
+         'abc  def\tghi'.split '\\s+' Regex_Matcher == ["abc", "def", "ghi"]
+Text.split : Text -> (Text_Matcher | Regex_Matcher) -> Vector.Vector Text
+Text.split delimiter="," matcher=Text_Matcher = case matcher of
+    Text_Matcher case_sensitivity -> case case_sensitivity of
+        True ->
+            Vector.Vector (Text_Utils.split_by_literal this delimiter)
+        Case_Insensitive locale ->
+            delimiters = Vector.Vector <|
+                Text_Utils.span_of_all_case_insensitive this delimiter locale.java_locale
+            current_start = Ref.new 0
+            acc = Vector.new_builder
```

Comment:
The Vector (Array) size is known beforehand so I wonder if better to do something like:
```
Vector.new delimiters.length+1 i->
    Text_Utils.substring this (if i == 0 then 0 else (delimiters.at i-1 . codeunit_end)) (delimiters.at i . codeunit_start)
```